docs/how-to-guides/troubleshoot.md

```diff
@@ -0,0 +1,174 @@
+ 
+**Root Cause:** The RootTemplateName field is required but is missing or empty.
+ 
+**Troubleshooting:** Examine the request body to ensure that the RootTemplateName field is present and is not null or empty.
+
+#### <u>InnerError: InvalidRequestBody</u>
+ 
+**Root Cause:** The request body failed validation for some reason other than those listed above.
+ 
+**Troubleshooting:** Examine the request body to ensure that all required fields are present and that the values are correctly formatted. If the request body appears to be correct, see the application logs for more details on why the request body failed validation.
+ 
+### InvalidTemplate
+ 
+The template content or name is invalid.
+ 
+#### <u>InnerError: InvalidFilter</u>
+ 
+**Root Cause:** An error was encountered during the convert operation while using a filter referenced by the conversion template.
+ 
+**Troubleshooting:** Examples of known causes of this error include usage of an invalid date-time format, an invalid hexadecimal number, and invalid time-zone handling. See the application logs for specific details of the error and examine any filters referenced by the conversion template to determine the source of the error. See more information on using filters [here](customize-templates.md). After addressing the issue in the template filter, upload the updated filter to the storage account, restart the container, and retry the request.
+ 
+#### <u>InnerError: InvalidTemplateContent</u>
+ 
+**Root Cause:** Some aspect of the template content is invalid.
+ 
+**Troubleshooting:** See the application logs for specific details of the error and examine the conversion template (both root template and templates referenced by the root template) to determine the source of the error. See more information on template authoring [here](customize-templates.md). After addressing the issue, upload the updated template to the storage account, restart the container, and retry the request.
+
+#### <u>InnerError: TemplateNotFound</u>
+ 
+**Root Cause:** The template name or path specified in the RootTemplateName field of the request body could not be found.
+ 
+**Troubleshooting:** Ensure that this value matches the value necessary to access the desired template. For default template requests, this should be only the template name. For example, to access the [ADT_A01](../../data/Templates/Hl7v2/ADT_A01.liquid) default template, the RootTemplateName field should be set to `ADT_A01`. For custom requests, this will be the name of the blob file containing the Liquid template in the storage account configured with the service. In the Azure portal, inspect your storage account to ensure that the provided `RootTemplateName` value matches the blob file name. Note that the `RootTemplateName` field should **not** contain the Storage Blob URI. See more information on writing valid request bodies to access custom templates in the [Use FHIR converter APIs](use-convert-web-apis) document.
```

Comment:
Mention that if they add a template to the storage account, they will need to restart the revision